### DISCUSSION TITLE
Query Batching not working as per our expectation

### DISCUSSION THREAD

**Discussion by Abhishek-Vig** [nhibernate/nhibernate-core#2736] [Q&A]
We are observing insert and update on different tables are not getting batched. We have both Session level(1000) and class leve(100) batching. Can anyone guide what can be the reason. Can anyone point to code where batching logic is written so that we can debug.

**Comment by fredericDelaporte** [comment]
There are many batching in NHibernate.
You should be more specific in your question and describe more specifically which operations are you doing, what is the behavior you are expecting, what is the behavior you are seeing.

The two main batching are lazy-loading batching and update or insert of entities batching.
You are likely asking about the later one. This batching batches together inserts of the same kind of entities (same class), or updates of the same kind of entities. It does not batch together inserts with updates, and it does not batch together operations done on different classes of entities. Dependencies among entities to be flushed may require an ordering of operations preventing batching. Insert baching can be hindered by the id generator strategy. By example, no batching will occur when saving entities using an identity generator, since each save will cause a flush to database.

**Reply by Abhishek-Vig** [reply]
Thanks for confirming. That's what we are observing inserts and updates are not getting batched and inserts on different entities not getting batched. Wanted to check whether we are missing some configuration.
